packages/main/src/Table.ts

```diff
@@ -420,6 +421,10 @@ class Table extends UI5Element {
 		this._renderNavigated = this.rows.some(row => row.navigated);
 		if (this.headerRow[0]) {
 			this.headerRow[0]._rowActionCount = this.rowActionCount;
+			if (this._getSelection()?.isMultiSelect() && this._rowsLength !== this.rows.length) {
```

Comment:
Just a minor remark:
We only need to invalidate the headerRow if the cached rowsLength is smaller than the actual row length, right?
But I guess just checking if they are not equal is the safer bet

Reply:
For now, what you’re saying is true, except for the "no data" case after selecting all. We can add this check as well, but we would need to extend it when handling the three-state checkbox. So, I’m not sure if trying to prevent this invalidation is really worth it.
Nevertheless, I can add the tests, thanks!